Update http.py

asterized opened the pull request:
Fixed problem in HTTPSession, renamed HTTPSession to _Session

asterized committed:
Update http.py

Fixed problem in HTTPSession, renamed HTTPSession to _Session

SomeHybrid commented:
list of fixes:
1: changed `self.close()` to `loop.run_until_complete(self.close())` as `aiohttp.ClientSession.close()` is an async function
2: initialized `aiohttp.ClientSession` in an async function to prevent DeprecationWarning as you need to initialize it in an async function

sexnine committed:
Formatting

SomeHybrid commented:
cool formatting

sexnine commented on bot/util/http.py:
```diff
@@ -61,22 +62,20 @@ async def inner(*args, no_cache=False, **kwargs):
+
+
+async def generate_session():
+    return _Session()
 
 
-session = HTTPSession()
+session = _loop.run_until_complete(generate_session())
```
This does not need to be async as initializing an object is sync.

```suggestion
session = _Session()
```

SomeHybrid replied:
initializing a ClientSession in a non-async function returns a DeprecationWarning

nuke886 replied:
`asyncio.get_event_loop()` was Deprecated in Python 3.10

sexnine reviewed (changes requested):
@nuke886 since your the one utilizing this class.  I'll leave you to deal with this, feel free to merge once your happy :)

asterized committed:
More formatting

More formatting (yay!)

SomeHybrid commented:
Details for more formatting as forgor to put it in the description

Added a Session class that returns _Session and i've made a mistake again sad

asterized committed:
apparently subclasses of ClientSession doesnt return the warning?

asterized committed:
more depreciation warning stuff

apparently putting it in a non async function is bad practice and also subclassing aiohttp.ClientSession is also bad practice
